homeassistant/components/swiss_public_transport/__init__.py

```diff
@@ -1 +1,38 @@
 """The swiss_public_transport component."""
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_DESTINATION, CONF_START, DEFAULT_NAME, DOMAIN
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_DESTINATION): cv.string,
+        vol.Required(CONF_START): cv.string,
+    }
+)
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+) -> bool:
+    """Set up platform from a ConfigEntry."""
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = entry.data
```

Comment:
Can we initialize the api client here, do a quick check if the service is working and then store it in the data?